app/models/security_log.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: security_log
+#
+#  id                    :bigint           not null, primary key
+#  action                :string(255)      not null
+#  details               :jsonb
+#  originator_id         :bigint
+#  originator_role       :string(255)
+#  originator_identifier :string(255)
+#  target_id             :integer
+#  target_type           :string(255)
+#  target_identifier     :string(255)
+#  logged_at             :datetime         not null
+#  originator_remote_ip  :string(255)
+#
+class SecurityLog < ApplicationRecord
+  self.table_name = 'security_log'
+
+  belongs_to :originator, class_name: 'User', optional: true
+
+  validates :action, presence: true, inclusion: { in: %w[
+    status_change account_update role_change accessed_site session_duration
+    create read update delete submit renewal_request
+  ] }
```

Comment:
these could be moved to a constant in case they're useful outside the class
```suggestion
  ROLES = %w[
    status_change account_update role_change accessed_site session_duration
    create read update delete submit renewal_request
  ].freeze
  validates :action, presence: true, inclusion: { in: ROLES }
```